### PR TITLE
BN-10 Create @Entity annotation and integrate process logic

### PR DESCRIPTION
**Pull request by rnyPlanet**

### CODE REVIEW

File: src/main/java/com/bobocode/svydovets/bibernate/session/SessionImpl.java [hunk 1]
```diff
@@ -14,6 +15,7 @@ public class SessionImpl implements Session {
 
     @Override
     public <T> T find(Class<T> type, Object id) {
+        EntityUtils.validateEntity(type);
```
Comment (ChonkaWork) [inline src/main/java/com/bobocode/svydovets/bibernate/session/SessionImpl.java:18]: Need to think how to apply validation before each method without putting it directly into methods. But right now it works for me

File: src/main/java/com/bobocode/svydovets/bibernate/util/EntityUtils.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        log.info("Validation %s".formatted(type.getName()));
+        isEntity(type);
+        //        todo: check that the class entity has at least one id
+        hasNoArgConstructor(type);
+    }
+
+    private static <T> void isEntity(Class<T> type) {
+        if (!type.isAnnotationPresent(Entity.class)) {
+            throw new EntityValidationException(
+                    "Class '"
```
Comment (ChonkaWork) [inline src/main/java/com/bobocode/svydovets/bibernate/util/EntityUtils.java]: Use String.format()
Also, add some logs here or in Exception class

File: src/test/java/com/bobocode/svydovets/bibernate/util/entities/EntityWithoutNonArgConstructor.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package com.bobocode.svydovets.bibernate.util.entities;
```
Comment (ChonkaWork) [inline src/test/java/com/bobocode/svydovets/bibernate/util/entities/EntityWithoutNonArgConstructor.java]: Mb move it to testdata package, or reuse existing classes

**Review by rovein — DISMISSED** [review]
Looks good, thanks!

File: src/main/java/com/bobocode/svydovets/bibernate/util/EntityUtils.java (hunk 2 above)
Comment (Prosperro-de) [inline src/main/java/com/bobocode/svydovets/bibernate/util/EntityUtils.java]: Could we use String.format() here?

File: src/test/java/com/bobocode/svydovets/bibernate/util/EntityUtilsTest.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    @Test
+    @DisplayName("Non-entity class validation")
+    void nonEntityClassValidation() {
+        EntityValidationException exception =
+                assertThrows(
+                        EntityValidationException.class,
+                        () -> EntityUtils.validateEntity(NonEntityClass.class));
+        assertThat(exception.getMessage())
+                .isEqualTo(
+                        "Class 'com.bobocode.svydovets.bibernate.util.entities.NonEntityClass' has no @Entity annotation (every entity class must be annotated with '@Entity')");
```
Comment (Prosperro-de) [inline src/test/java/com/bobocode/svydovets/bibernate/util/EntityUtilsTest.java]: Let's create a constant for this

File: src/main/java/com/bobocode/svydovets/bibernate/util/EntityUtils.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+import com.bobocode.svydovets.bibernate.exception.EntityValidationException;
+import java.util.Arrays;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class EntityUtils {
+    private EntityUtils() {}
+
+    public static <T> void validateEntity(Class<T> type) {
+        log.info("Validation %s".formatted(type.getName()));
```
Comment (Prosperro-de) [inline src/main/java/com/bobocode/svydovets/bibernate/util/EntityUtils.java]: I beliave logger supports string formating with {} for instance `log.info("Validation {}", type.getName());`

File: src/test/java/com/bobocode/svydovets/bibernate/util/EntityUtilsTest.java (hunk 4 above)
Comment (Prosperro-de) [reply src/test/java/com/bobocode/svydovets/bibernate/util/EntityUtilsTest.java]: And also for other strings

Comment (rnyPlanet) [reply src/test/java/com/bobocode/svydovets/bibernate/util/EntityUtilsTest.java]: @Prosperro-de where will we store them?

Comment (Prosperro-de) [reply src/test/java/com/bobocode/svydovets/bibernate/util/EntityUtilsTest.java]: @rnyPlanet We could have a common class with constants with %s placeholders that we will use in exceptions and pass the class.getName values here as well

**Review by rovein — DISMISSED** [review]

**Review by rovein — DISMISSED** [review]

**Review by rovein — APPROVED** [review]

**Review by ChonkaWork — APPROVED** [review]

**Review by Prosperro-de — APPROVED** [review]

**Pull request merged**